extensions/functions_string.yaml

```diff
@@ -88,7 +88,10 @@ scalar_functions:
       from the beginning of the string to begin starting to search for pattern matches can be
       specified using the `position` argument. Specifying `1` means to search for matches
       starting at the first character of the input string, `2` means the second character, and so
-      on. The `position` argument should be a positive non-zero integer.
+      on. The `position` argument should be a positive non-zero integer. The regular
+      expression capture group can be specified using the `group` argument. Specifying `0`
+      will match the entire regular expression.  The default group value is `1`, which means the
+      first capture group will be matched. The `group` argument should be a non-negative integer.
```

Comment:
@richtia thank you for adding this `group` argument to the existing `regexp_match_substring` function. This creates better correspondence between this existing function and the new `regexp_match_substring_all` function added here. But to be clear, it's not only for the sake of completeness. Multiple real-world functions that extract a single regex match—including R's [`stringr::str_extract()`](https://stringr.tidyverse.org/reference/str_extract.html) and Snowflake's with [`REGEXP_SUBSTR`](https://docs.snowflake.com/en/sql-reference/functions/regexp_substr)—accept an optional group number argument which allows the user to specify which capture group to return. So this is a practical addition.

Is it considered a breaking change when we add a new argument like this? I forget whether or not Substrait technically allows optional arguments / arguments with default values.

Reply:
I think it would be considered a breaking change.  I'm not 100% sure though.  @westonpace

Reply:
Yes, adding a new positional argument is a breaking change (adding a new option is not).